hudi-hadoop-common/src/main/java/org/apache/hudi/io/hadoop/HoodieBaseParquetWriter.java

```diff
@@ -146,8 +147,25 @@ public long getDataSize() {
   }
 
   public void write(R object) throws IOException {
-    this.parquetWriter.write(object);
-    writtenRecordCount.incrementAndGet();
+    try {
+      this.parquetWriter.write(object);
+      writtenRecordCount.incrementAndGet();
+    } catch (RuntimeException e) {
+      String errorMessage = e.getMessage() != null ? e.getMessage() : "";
+      if (isRequiredFieldNullError(errorMessage) && errorMessage.contains("_hoodie_is_deleted")) {
```

Comment:
Where does these exceptions come from?  Could `errorMessage.contains` be avoided and the source to throw an exception of specific classes which can be caught here?

Reply:
guess we can avoid this fix. exiting error msg 
```
Null-value for required field: _hoodie_is_deleted
```

would suffice. I just don't feel intercepting at this layer just to add more enhancement to error msg is not worth it